app/grandchallenge/components/utils.py

```diff
@@ -0,0 +1,49 @@
+import json
+
+from grandchallenge.components.models import InterfaceKindChoices
+from grandchallenge.core.validators import JSONValidator
+from grandchallenge.hanging_protocols.models import (
+    VIEW_CONTENT_SCHEMA,
+    ViewportNames,
+)
+
+
+def generate_view_content_example(interfaces):
+    images = list(
+        interfaces.filter(kind=InterfaceKindChoices.IMAGE)
+        .order_by("slug")
+        .values_list("slug", flat=True)
+    )
+
+    if not images:
+        return None
+
+    overlays = list(
+        interfaces.filter(
+            kind__in=[
+                InterfaceKindChoices.SEGMENTATION,
+                InterfaceKindChoices.HEAT_MAP,
+                InterfaceKindChoices.DISPLACEMENT_FIELD,
+            ]
+        )
+        .order_by("slug")
+        .values_list("slug", flat=True)
+    )
+    view_content_example = {}
+    overlays_per_image = len(overlays) // len(images)
+    remaining_overlays = len(overlays) % len(images)
+
+    for port in ViewportNames.values:
+        if len(images) == 0:
+            break
+
+        view_content_example[port] = [images.pop(0)]
+        for _ in range(overlays_per_image):
+            view_content_example[port].append(overlays.pop(0))
+        if remaining_overlays > 0:
+            view_content_example[port].append(overlays.pop(0))
+            remaining_overlays -= 1
+
+    JSONValidator(schema=VIEW_CONTENT_SCHEMA)(value=view_content_example)
```

Comment:
I don't think checking the example for adherence to the schema here makes sense. It should always be valid. If it's not valid, this will raise an error and will make the entire form unusable for the user. I don't think that's what we want. 

Ultimately, what we need is to make sure that the generated example adheres to the schema and that it continues to do so even when the schema changes. To achieve that, I think a test that calls this function and validates it against the schema is the way to go. Future developers will then be notified if their changes to the view content schema break this function and can update the code accordingly.

Reply:
A test is necessary but I don't think that it is possible to cover all situations a-priori. In that case, you still need a runtime check. Having an un-rendered view is okay, but it is useful to catch the `ValidationError` here and raise it as a `RuntimeError` so that future developers know from Sentry that this is not actually a data validation problem, but that the logic in this method that would need to be fixed.

Reply:
Actually, after confirming that all interfaces need to be present in the example, I think the generated example should go through the same validation path as the `view_content` field in the [HangingProtocolMixin](https://github.com/comic/grand-challenge.org/blob/9c7930c1a7d2c28af4753dbaac249931eae855a3/app/grandchallenge/hanging_protocols/models.py#L305). That is, checking that `isolated_interfaces` are solely present in a viewport, only one `image` in a view port, and no `undisplayable_interfaces` present in any view port.

However, since the `HangingProtocolMixin` is an Abstract class, it **can not** be instantiated. So, I **can not** do:
```
hp = HangingProtocolMixin(view_content=view_content_example)
hp.full_clean()

```
I am exploring alternatives.